src/transformState.js

```diff
@@ -5,7 +5,21 @@
  * @param {Object[]} actions
  */
 function transformState(state, actions) {
-  // write code here
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        Object.assign(state, action.extraData);
+        break;
+      case 'removeProperties':
+        action.keysToRemove.map(key => delete state[key]);
+        break;
+      case 'clear':
+        Object.keys(state).map(key => delete state[key]);
+        break;
```

Comment:
we mutate our initial state let's prevent this behavior and just return new state from function

cloning objects in js- [link](https://www.freecodecamp.org/news/clone-an-object-in-javascript/)

Reply:
![image](https://github.com/mate-academy/js_stateful-object/assets/95931150/f1b727a3-1fac-40d1-a937-1311664c2503)